### ISSUE TITLE
'x86_64-linux' target filtering unexpectedly returns other targets like windows

### ISSUE THREAD

**Issue by gavindidrichsen** [habitat-sh/builder#1418]
## The Enhancement:

I want to be able to filter a list on a target like x86-linux for a curl request to obtain all package identifiers on a channel.  Currently this endpoint, which is detailed later:
* returns all packages and targets for a particular channel
* does not distinguish the 'target' for each pkg_ident

There are many ways that this could be done:  (1) add the 'target to the json response object for each pkg_ident?  (2) add an overall metadata tag at the level of the 'data' element that defines what target, channel, origin.  Something like:

```diff
--- a/scripts/on_prem_builder/test/habitat/utils/fixtures/expected/list_of_pkgidents/given/curl_responses/curl_output_0.json
+++ b/scripts/on_prem_builder/test/habitat/utils/fixtures/expected/list_of_pkgidents/given/curl_responses/curl_output_0.json
@@ -31,7 +35,9 @@
       "origin": "core",
       "name": "aws-cli",
       "version": "1.18.25",
       "release": "20200319204505",
+      "target": "x86-linux",
+      "channel": "refresh-2020q1-008"
     },
     {
       "origin": "core",
```

or

```diff
--- a/scripts/on_prem_builder/test/habitat/utils/fixtures/expected/list_of_pkgidents/given/curl_responses/curl_output_0.json
+++ b/scripts/on_prem_builder/test/habitat/utils/fixtures/expected/list_of_pkgidents/given/curl_responses/curl_output_0.json
@@ -2,6 +2,10 @@
   "range_start": 0,
   "range_end": 49,
   "total_count": 116,
+  "metadata": {
+    "target": "x86-linux",
+    "channel": "refresh-2020q1-008"
+  },
   "data": [
     {
       "origin": "core",
```

### The Need:

This would be useful especially when attempting to verify the change or delta to a channel that has occurred.  For example, 
* I've uploaded 50 packages to my channel, how do I verify that the exact identifiers have been successfully uploaded?  
* What's changed on the stable channel for the x86-linux target between yesterday and today?  

Being able to download yesterday's and today's full list allows a delta to be computed 

### Current Alternative

There doesn't seem to be an alternative that I know of.  However, this is not blocking us.  I do have a workaround at the minute which is to (1) download the full list (2) normalize the list by stripping off the version/timestamp and uniquify; (3) use the unique list and do individual curl's to Builder to obtain the 'latest' version for each.

## Context

Notice the following curl is filtering on the 'x86_64-linux' target:
```bash
curl -s --insecure -H 'Accept: application/json' 'https://bldr.habitat.sh/v1/depot/channels/chef/stable/pkgs?target=x86_64-linux&range=500'
```

This returns:
```json
'{"range_start":500,"range_end":549,"total_count":661,"data":[{"origin":"chef","name":"ruby27-plus-devkit","version":"2.7.0","release":"20200423155117"},{"origin":"chef","name":"ruby-plus-devkit","version":"2.6.5","release":"20191017152652"},{"origin":"chef","name":"scaffolding-chef","version":"0.5.0","release":"20190621191240"},{"origin":"chef","name":"scaffolding-chef-benchmark","version":"0.1.0","release":"20200504115315"},{"origin":"chef","name":"scaffolding-chef-benchmark","version":"0.1.0","release":"20200504115319"},{"origin":"chef","name":"scaffolding-chef-benchmark","version":"0.1.0","release":"20200504214814"},{"origin":"chef","name":"scaffolding-chef-benchmark","version":"0.1.0","release":"20200504214817"},{"origin":"chef","name":"scaffolding-chef-benchmark","version":"0.1.0","release":"20200505140456"},{"origin":"chef","name":"scaffolding-chef-benchmark","version":"0.1.0","release":"20200505140637"},{"origin":"chef","name":"scaffolding-chef-benchmark","version":"0.1.0","release":"20200509220602"},{"origin":"chef","name":"scaffolding-chef-benchmark","version":"0.1.0","release":"20200509220607"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.10.0","release":"20190718003851"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.10.0","release":"20190718003905"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.10.0","release":"20190718004025"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.10.0","release":"20190718132704"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.10.0","release":"20190718132829"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.10.0","release":"20190718132935"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.11.0","release":"20190724153035"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.11.0","release":"20190724153117"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.11.0","release":"20190724154443"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.12.0","release":"20190724225917"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.12.0","release":"20190724225918"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.12.0","release":"20190724230839"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.13.0","release":"20190801172202"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.13.0","release":"20190801172329"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.13.0","release":"20190801172343"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.14.0","release":"20190924192426"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.14.0","release":"20190924192608"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.14.0","release":"20190924193159"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.15.0","release":"20191002135721"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.15.0","release":"20191002135907"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.15.0","release":"20191002140439"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.16.0","release":"20191028150428"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.16.0","release":"20191028150617"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.16.0","release":"20191028151207"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.16.0","release":"20191219205349"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.16.0","release":"20191219205548"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.16.0","release":"20191219205819"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.16.0","release":"20191219224458"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.16.0","release":"20191219224721"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.16.0","release":"20191219224900"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.17.0","release":"20200206175728"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.17.0","release":"20200206175857"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.17.0","release":"20200206180517"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.17.0","release":"20200206233121"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.17.0","release":"20200206233318"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.17.0","release":"20200206233932"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.18.0","release":"20200213132428"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.18.0","release":"20200213132439"},{"origin":"chef","name":"scaffolding-chef-infra","version":"0.18.0","release":"20200213132614"}]}'
```

Which contains ``chef/ruby27-plus-devkit/2.7.0/20200423155117``

This was not an expected result since chef/ruby27-plus-devkit only has one Windows version on Live Builder

<img width="1178" alt="image" src="https://user-images.githubusercontent.com/16652757/81555444-7cf8d280-9380-11ea-964f-69ea59c65302.png">

**Comment by jeremymv2** [comment]
@gavindidrichsen we don't currently support usage of the `target` query parameter on this API endpoint.

It is, however, supported on these endpoints:
`/depot/channels/{origin}/{channel}/pkgs/{pkg}/latest`
`/depot/channels/{origin}/{channel}/pkgs/{pkg}/{version}/latest`
`/depot/channels/{origin}/{channel}/pkgs/{pkg}/{version}/{release}`
`/depot/channels/{origin}/{channel}/pkgs/{pkg}/{version}/{release}/promote`
`/depot/channels/{origin}/{channel}/pkgs/{pkg}/{version}/{release}/demote`

**Comment by stevendanna** [comment]
Perhaps this should become a feature request? There is a good deal of tooling that would be a lot easier to write if target applied to the `pkgs` endpoint.

**Comment by jeremymv2** [comment]
@gavindidrichsen if you could take a few moments to submit the criteria we use for feature requests, it will help us understand how to prioritize any work.

### Describe the Enhancement:
[What you are trying to achieve that you can't?]

### Describe the Need:
[What kind of user do you believe would utilize this enhancement, and how many users might want this functionality]

### Current Alternative
[Is there a current alternative that you can utilize to workaround the lack of this enhancement or are you currently blocked?]

Thank you!

**Comment by gavindidrichsen** [comment]
Hi @jeremymv2.  I've updated the issue with the feature request details.  Cheers.

**Comment by gavindidrichsen** [comment]
Hi @jeremymv2.  I've updated the enhancement request--see above.

**Comment by jeremymv2** [comment]
closed via #1419
